app/screens/settings/notification_mention/mention_settings.tsx

```diff
@@ -63,48 +49,119 @@ const getStyleSheet = makeStyleSheetFromTheme((theme) => {
+    componentId: string;
+    currentUser: UserModel;
 }
-const MentionSettings = ({currentUser, mentionKeys}: MentionSectionProps) => {
-    const [{firstName, usernameMention, channel}, dispatch] = useReducer(reducer, INITIAL_STATE);
+const MentionSettings = ({componentId, currentUser}: MentionSectionProps) => {
+    const serverUrl = useServerUrl();
+
+    const notifyProps = currentUser.notifyProps as Partial<UserNotifyProps>;
+    const [tglFirstName, setTglFirstName] = useState(notifyProps.first_name === 'true');
+    const [tglChannel, setTglChannel] = useState(notifyProps.channel === 'true');
+
+    const mentionProps = useMemo(() => getMentionProps(currentUser), [currentUser.notifyProps]);
+    const [tglUserName, setTglUserName] = useState(mentionProps.usernameMention);
+    const [mentionKeys, setMentionKeys] = useState(mentionProps.mentionKeys);
+
     const theme = useTheme();
     const styles = getStyleSheet(theme);
     const intl = useIntl();
 
-    const toggleChannelMentions = () => {
-        dispatch({
-            type: UPDATE_MENTION_PREF,
-            data: {
-                channel: !channel,
-            },
-        });
-    };
-    const toggleUsernameMention = () => {
-        dispatch({
-            type: UPDATE_MENTION_PREF,
-            data: {
-                usernameMention: !usernameMention,
-            },
-        });
-    };
-    const toggleFirstNameMention = () => {
-        dispatch({
-            type: UPDATE_MENTION_PREF,
-            data: {
-                firstName: !firstName,
-            },
+    const saveButton = useMemo(() => {
+        return {
+            id: SAVE_MENTION_BUTTON_ID,
+            enabled: false,
+            showAsAction: 'always' as const,
+            testID: 'notification_settings.mentions.save.button',
+            color: theme.sidebarHeaderTextColor,
+            text: intl.formatMessage({id: 'settings.save', defaultMessage: 'Save'}),
+        };
+    }, [theme.sidebarHeaderTextColor]);
+
+    const canSave = useCallback(() => {
+        const fNameChanged = tglFirstName !== Boolean(notifyProps.first_name);
+        const channelChanged = tglChannel !== Boolean(notifyProps.channel);
+
+        const usnChanged = tglUserName !== mentionProps.usernameMention;
+        const kwsChanged = mentionProps.mentionKeys !== mentionKeys;
+
+        const enabled = fNameChanged || usnChanged || channelChanged || kwsChanged;
+
+        const buttons = {
+            rightButtons: [{
+                ...saveButton,
+                enabled,
+            }],
+        };
+        setButtons(componentId, buttons);
+    }, [componentId, saveButton, tglFirstName, tglChannel, tglUserName, mentionKeys, notifyProps]);
+
+    const onChangeText = useCallback((text: string) => {
+        setMentionKeys(text);
+        canSave();
+    }, [canSave]);
+
+    const close = useCallback(() => popTopScreen(componentId), [componentId]);
+
+    const saveMention = useCallback(() => {
```

Comment:
I don't think this needs to be memoized, thoughts?

Reply:
The way it is being used, it does not need to be memoized. `useNavButtonPressed` only uses the dependency list to re-render, so passing a new `saveMention` on every render won't do much, and we are passing in the dependencies all relevant values that trigger this.

All that being said, for sanity sake, I see value on having this function memoized, and then have only this function in the `useNavButtonPressed` dependencies.

Reply:
corrected